From 33ef2b9ac886fd2e37f77712ac1adaec078fc505 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Tue, 1 Sep 2015 13:29:18 +0800
Subject: [PATCH 0363/5242] MLK-11471-02 ARM: dts: i.mx6q/dl/sx: Add isl29023
 device tree support.

commit  3048fec5368afbda745b695bfb2e457d395bf65a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add isl29023 sensor device tree support on i.MX6Q/DL/SX platform.

Signed-off-by: Luwei Zhou <b45643@freescale.com>
Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabreauto.dtsi |    9 +++++++++
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi   |   10 ++++++++++
 arch/arm/boot/dts/imx6sx-sdb.dtsi        |    9 +++++++++
 3 files changed, 28 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
index ecc01af..8db1c65 100644
--- a/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabreauto.dtsi
@@ -409,6 +409,14 @@
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
 
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		interrupt-parent = <&gpio5>;
+		interrupts = <17 2>;
+	};
+
 	mag3110@0e {
 		compatible = "fsl,mag3110";
 		reg = <0x0e>;
@@ -447,6 +455,7 @@
 				MX6QDL_PAD_GPIO_18__SD3_VSELECT 0x17059
 				MX6QDL_PAD_EIM_BCLK__GPIO6_IO31 0x80000000
 				MX6QDL_PAD_EIM_EB1__GPIO2_IO29	0x80000000
+				MX6QDL_PAD_DISP0_DAT23__GPIO5_IO17 0x80000000
 			>;
 		};
 
diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 7f174a8..0ff03c6 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -432,6 +432,15 @@
 		wakeup-gpios = <&gpio6 7 0>;
 	};
 
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		vdd-supply = <&reg_sensor>;
+		interrupt-parent = <&gpio3>;
+		interrupts = <9 2>;
+	};
+
 	mag3110@0e {
 		compatible = "fsl,mag3110";
 		reg = <0x0e>;
@@ -462,6 +471,7 @@
 				MX6QDL_PAD_EIM_EB3__GPIO2_IO31 0x80000000
 				MX6QDL_PAD_SD1_CMD__GPIO1_IO18 0x80000000
 				MX6QDL_PAD_EIM_D16__GPIO3_IO16 0x80000000
+				MX6QDL_PAD_EIM_DA9__GPIO3_IO09 0x80000000
 			>;
 		};
 
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index 92e3c45..b7b2373 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -208,6 +208,15 @@
 	pinctrl-0 = <&pinctrl_i2c3>;
 	status = "okay";
 
+	isl29023@44 {
+		compatible = "fsl,isl29023";
+		reg = <0x44>;
+		rext = <499>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <5 1>;
+		shared-interrupt;
+	};
+
 	mag3110@0e {
 		compatible = "fsl,mag3110";
 		reg = <0x0e>;
-- 
1.7.9.5

